logs/client.go

```diff
@@ -0,0 +1,200 @@
+package logs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"math/rand/v2"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/solarwinds/swo-cli/version"
+)
+
+type ColorStrFunc func(format string, a ...interface{}) string
+
+var colors = []ColorStrFunc{
+	color.CyanString,
+	color.YellowString,
+	color.GreenString,
+	color.MagentaString,
+	color.RedString,
+}
+
+type Client struct {
+	opts       *Options
+	httpClient http.Client
+	output     *os.File
+}
+
+type Log struct {
+	Time     time.Time `json:"time"`
+	Message  string    `json:"message"`
+	Hostname string    `json:"hostname"`
+	Severity string    `json:"severity"`
+	Program  string    `json:"program"`
+}
+
+type PageInfo struct {
+	PrevPage string `json:"prevPage"`
+	NextPage string `json:"nextPage"`
+}
+
+type LogsData struct {
+	Logs     []Log `json:"logs"`
+	PageInfo `json:"pageInfo"`
+}
+
+func NewClient(opts *Options) (*Client, error) {
+	return &Client{
+		httpClient: *http.DefaultClient,
+		opts:       opts,
+		output:     os.Stdout,
+	}, nil
+}
+
+func (c *Client) prepareRequest(ctx context.Context) (*http.Request, error) {
+	logsEndpoint, err := url.JoinPath(c.opts.ApiUrl, "v1/logs")
+	if err != nil {
+		return nil, err
+	}
+
+	logsUrl, err := url.Parse(logsEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	params := url.Values{}
+	params.Add("pageSize", strconv.Itoa(c.opts.count))
+	if c.opts.group != "" {
+		params.Add("group", c.opts.group)
+	}
+	if c.opts.minTime != "" {
+		params.Add("startTime", c.opts.minTime)
+	}
+	if c.opts.maxTime != "" {
+		params.Add("endTime", c.opts.maxTime)
+	}
+
+	var filter string
+	if c.opts.system != "" {
+		filter = fmt.Sprintf("host:%s", c.opts.system)
+	}
+	if len(c.opts.args) != 0 {
+		if len(filter) == 0 {
+			filter = strings.Join(c.opts.args, " ")
+		} else {
+			filter = filter + " " + strings.Join(c.opts.args, " ")
+		}
+	}
+
+	if filter != "" {
+		params.Add("filter", filter)
+	}
+
+	logsUrl.RawQuery = params.Encode()
+	request, err := http.NewRequestWithContext(ctx, "GET", logsUrl.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.opts.Token))
+	request.Header.Add("Accept", "application/json")
+
+	return request, nil
+}
+
+func (c *Client) printResult(logs *LogsData) error {
+	if c.opts.json {
+		jsonFormat, err := json.Marshal(logs)
+		if err != nil {
+			return err
+		}
+
+		_, err = fmt.Fprintln(c.output, string(jsonFormat))
+		return err
+	}
+
+	hostnameColorIdx := -1
+	programColorIdx := -1
+	switch c.opts.color {
+	case system:
+		hostnameColorIdx = rand.IntN(len(colors))
+	case program:
+		programColorIdx = rand.IntN(len(colors))
+	case all:
+		programColorIdx = rand.IntN(len(colors))
+		hostnameColorIdx = rand.IntN(len(colors))
+		for hostnameColorIdx == programColorIdx {
+			hostnameColorIdx = rand.IntN(len(colors))
+		}
+	default:
+	}
```

Comment:
The Papertrail CLI uses colors a bit differently here. It's not trying to mimic the Log Viewer UI, but instead it shows each different program (or each host, or each host+program, depending on the option selected here) in a different color. This is meant to help recognie where the logs are coming from, at a glance.

Also, this feature probably doesn't need to exist in V1.

Reply:
Ok, deleted for now. We might do this in future.